src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -1117,25 +1117,45 @@ module SimulationCode =
             | _  ->
                 ``property-get`` "System.Collections.Generic.IEnumerable<Qubit>" "IApplyData.Qubits" [] 
                     ``get`` (fieldPaths |> List.map buildOne)
-            :> MemberDeclarationSyntax
         else
-            let buildOne (field: ResolvedType * ExpressionSyntax) = 
-                let t = field |> fst 
+            let mutable count = 0
+            let nextName() =
+                count <- count + 1
+                sprintf "__temp%d__" count
+            let mutable items = []
+            for (t, token) in fields do 
                 match t.Resolution with
                 | QsTypeKind.Function _
                 | QsTypeKind.Operation _
                 | QsTypeKind.ArrayType _
                 | QsTypeKind.UserDefinedType _
-                | QsTypeKind.Qubit -> ``((`` ( ``cast`` "IApplyData" (field |> snd) ) ``))`` <|?.|> ``ident`` "Qubits"
-                | _       -> (field |> snd)  <?.>  ( ``ident`` "GetQubits", [] )
-            let body = 
+                | QsTypeKind.Qubit -> 
+                    let qs = ``((`` ( ``cast`` "IApplyData" token) ``))`` <|?.|> ``ident`` "Qubits"
+                    items <- (null, qs) :: items
+                | _       -> 
+                    let id = nextName()
```

Comment:
If I was reading this without knowing about the .NET Core bug, this would seem pretty awkward to me. Perhaps would it be reasonable to add a comment here indicating the workaround?

Reply:
Sure, done.